### ISSUE TITLE
📦 ONNX | Export Phi-3-Power as Polynomial-Only Graph

### ISSUE THREAD

**Issue by collinbentley1** [medlock-ai/medlock#27]
### Purpose
EncryptedLLM's CKKS compiler ingests ONNX where **all non-linearities are low-degree polynomials**.  
We must export the full model, replacing incompatible ops, and ensure numerical parity.

### Steps
1. **Activation substitutions**
   | Original | Replacement | Degree | File |
   |----------|-------------|--------|------|
   | GeLU     | Chebyshev degree-3 (T₁,T₃ terms) | 3 | `layers/cheb_gelu.py` |
   | LayerNorm| PolyNorm variant (degree-2)      | 2 | `layers/poly_norm.py` |

2. **Write Export Script**
   ```
   python scripts/export_poly_onnx.py \
     --pt_checkpoint ./weights/phi3_power \
     --out models/phi3_poly/phi3_poly.onnx \
     --opset 17
   ```
   Key points:  
   * Insert `Linear` → `MatMul + Add` nodes (ONNX requirement).  
   * Attach custom attributes: `poly_degree=3` for Cheb nodes.  
   * Convert weight RT to FP16 to shrink size.

3. **Graph optimisation pass**
   * Use `onnx-simplifier` then `onnxruntime_tools.optimizer` to fuse Gemm.

4. **Plaintext sanity test**
   * `tests/test_poly_onnx_equivalence.py`  
     * Load ONNX via `onnxruntime.InferenceSession` (CPU).  
     * Compare logits on 20 random prompts vs PyTorch baseline; assert MSE < 1e-3.

5. **Artifact publishing**
   * Add LFS pointer for ONNX; file must be < 2.8 GB.  
   * Provide SHA256 checksum in `models/README.md`.

6. **CI**
   * Workflow `onnx_validation.yml` to run equivalence test on small runner.

### Acceptance Criteria
- `phi3_poly.onnx` exists; validated equivalence tests green.
- No unsupported ops when running `onnxruntime_tools.optimizer.check_graph`.
- `models/README.md` documents replacement formulas.
